packages/apputils/src/sessioncontext.tsx

```diff
@@ -1484,21 +1510,290 @@ export class SessionContextDialogs implements ISessionContext.IDialogs {
   private _settingRegistry: ISettingRegistry | null;
 }
 
+export namespace SessionContextDialogs {
+  type KernelOptions = {
+    disabled?: boolean;
+    groups: Array<KernelOptionGroup>;
+  };
+
+  type KernelOption = {
+    selected?: boolean;
+    text: string;
+    title?: string;
+    value: string;
+  };
+
+  type KernelOptionGroup = {
+    label: string;
+    options: Array<KernelOption>;
+  };
```

Comment:
Since `KernelOptions` is returned by an exported function, should it be exported too, and by extension, should the other two be exported and documented too?

Should these by types or interfaces? Types can get annoying in TS when working with multiple versions of dependencies.

Reply:
I'm happy to change them to interfaces.

I don't have strong feelings regarding whether they should be public. I think maybe keeping them private is slightly preferable because these aren't really types we expect others to create and a smaller public footprint is nicer.

Do you think they ought to be public interfaces?

Reply:
I moved everything into one public `interface`.